### PR TITLE
Update Connector SDK

### PR DESCRIPTION
**Pull request by lovromazgon**
### Description

Updates connector SDK to latest version in `main`.

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-postgres/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

**Review by hariso — APPROVED** [review]
:elephant: :rocket:

**Comment by lovromazgon** [comment]
Let's before we merge this, as the connector is actively being developed. We should merge right before releasing the new SDK and Conduit version.

**Pull request merged**
